flake.nix

```diff
@@ -0,0 +1,169 @@
+{
+  description = "Shadysim";
+
+  inputs = {
+
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-21.05";
+
+    pycrypto-src = {
+      url = "github:pycrypto/pycrypto";
+      flake = false;
+    };
+
+    simtools-src = {
+      url = "git+https://git.osmocom.org/sim/sim-tools";
+      flake = false;
+    };
+
+    imsi-pseudo-src = {
+      url = "git+https://git.osmocom.org/imsi-pseudo";
+      flake = false;
+    };
+
+  };
+
+
+  outputs =
+    { self
+    , nixpkgs
+    , pycrypto-src
+    , simtools-src
+    , imsi-pseudo-src
+    , ...
+    }:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }
+      );
+
+    in
+    {
+
+      overlay = final: prev: rec {
+
+        pycrypto =
+          with final.python2Packages;
+          buildPythonPackage {
+            pname = "pycrypto";
+            version = "2.6.1";
+            src = pycrypto-src;
+          };
+
+        shadysim =
+          { python2
+          , python2Packages
+          }:
+
+            with python2Packages;
+            buildPythonApplication {
+              name = "shadysim";
+              version = "0.1.0";
+              src = "${simtools-src}/shadysim";
+
+              patches = [
+                ./patches/0001-add-setup-py.patch
+              ];
+
+              postInstall = ''
+                mv $out/bin/shadysim{.py,}
+              '';
+
+              propagatedBuildInputs = [
+                pycrypto
+              ];
+            };
+
+        shadysim-bin = with final; callPackage shadysim { };
```

Comment:
if you only use shadysim here, why did you separate shadysim ?
Would it be possible to just one shadysim-bin including everything ?

Reply:
Yep, lets do that. I have removed all `-bin`s and now we have just `shadysim` and `converter`, which are both user facing packages.